src/e2e-test/java/io/cdap/plugin/odp/stepsdesign/DesginTimeODP.java

```diff
@@ -0,0 +1,135 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.odp.stepsdesign;
+
+import io.cdap.e2e.pages.actions.CdfGcsActions;
+import io.cdap.e2e.utils.SeleniumDriver;
+import io.cdap.e2e.utils.SeleniumHelper;
+import io.cdap.plugin.odp.actions.ODPActions;
+import io.cdap.plugin.odp.locators.ODPLocators;
+import io.cdap.plugin.odp.utils.CDAPUtils;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import org.apache.commons.lang3.StringUtils;
+import org.junit.Assert;
+import org.openqa.selenium.By;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.io.IOException;
+
+/**
+ * DesginTimeODP.
+ */
+public class DesginTimeODP {
+
+  private static final String A = "a";
+
+  @Then("Connection is established")
+  public void connectionIsEstablished() throws InterruptedException {
+    CdfGcsActions.getSchema();
+    WebDriverWait wait = new WebDriverWait(SeleniumDriver.getDriver(), 20000);
+    wait.until(ExpectedConditions.numberOfElementsToBeMoreThan(By.xpath("//*[@placeholder=\"Field name\"]"), 2));
+    Assert.assertEquals(true, CDAPUtils.schemaValidation());
+  }
+
+  @When("LoadProp {string} {string} {string} {string} {string} {string} {string} {string} {string}")
+  public void loadConnection(String client, String asHost, String msServ, String systemID, String dsName,
+                             String gcsPath, String splitRow, String pkgSize, String lgrp)
+    throws IOException, InterruptedException {
+    ODPActions.getODPProperties();
+    ODPActions.enterLoadConnectionProperties(
+      client, asHost, msServ, systemID, lgrp, dsName, gcsPath, splitRow, pkgSize);
+  }
+
+  @When("User has selected Sap client macro to configure")
+  public void userHasSelectedSapClientMacroToConfigure() {
+    ODPLocators.macros.get(0).click();
+    ODPLocators.sapClient.sendKeys(Keys.chord(Keys.CONTROL, A));
+    ODPLocators.sapClient.sendKeys("${client}");
+  }
+
+  @Then("User is validate without any error")
+  public void userIsValidateWithoutAnyError() throws InterruptedException {
+    ODPLocators.validateButton.click();
+    Assert.assertTrue(ODPLocators.successMessage.isDisplayed());
+  }
+
+  @When("User has selected Sap language macro to configure")
+  public void userHasSelectedSapLagMacroToConfigure() {
+    ODPLocators.macros.get(1).click();
+    ODPLocators.language.sendKeys(Keys.chord(Keys.CONTROL, A));
+    ODPLocators.language.sendKeys("${lang}");
+  }
+
+  @When("User has selected Sap server as host macro to configure")
+  public void userHasSelectedSapServerAsHostMacroToConfigure() {
+    ODPLocators.macros.get(2).click();
```

Comment:
Is there any better way to handle this? This makes the test brittle. In future, if the plugin configs order is changed then whole test will break! @rmstar What do you think?

Reply:
@itsankit-google I dont think so we can handle it any other way as these macros are in a single class only the index is the difference. If order changes then we need to again change the order of identification

Reply:
yeah seems brittle.. what happens if a new config param is added? would that affect the index of other macros?
would be good if we can find a better way to handle it.

Reply:
got the solution for every macro element we can capture an object path will remain constant- //*[@data-cy="jco.client.ashost"]//child::button//span//span//strong only the data-cy value will be changed. @sumitnigamcs please capture all the macros seprately and call via name

Reply:
done